### ISSUE TITLE
PutObjectAsync is not always successful

### ISSUE THREAD

**Issue by ToBeMyOwnHero** [aws/aws-sdk-cpp#1494]
When I try to call the _PutObjectAsync_  several times in a short time, but the result is not always successful.Example, I call the _PutObjectAsync_ 1000 times in a short time, but sometimes I just got 100 put-object-request in the packets which was catched by tcpdump.
So, I really want to know the reason.Or, you can give me some ideas to find the reason, which like how to use and find the debug log of SDK.

**Comment by KaibaLopez** [comment]
Hi @ToBeMyOwnHero ,
I guess depending on how short a "short time" is, and if by just got 100 you mean 900 failed or that you got 100 at a time.
Anyways, to activate debug logs add:
```
Aws::SDKOptions options;
options.loggingOptions.logLevel = Aws::Utils::Logging::LogLevel::Debug;
Aws::InitAPI(options);
...
Aws::ShutdownAPI(options);
```
Also, this information is part of the github issue templates we provide, please use one of those when opening a new issue.

**Comment by ToBeMyOwnHero** [comment]
When I called the _PutObjectAsync_  every 50 subtle by using the _PooledThreadExecutor_ , I saw that the  thread was not always successful.In this case, I called the _PutObjectAsync_ 1000 times, effectively, I sent put-object-request less than 1000 times and just 100 times to 300 times.Dose this matter with the m_poolSize? I set the m_poolSize to 100. And, the debug log dose not record this case. By the log, I even don't konw anything about ThreadPool. I just got how many requset has been sent, but can't to konw why the other requests has not been sent.
